gnc/sub8_trajectory_generator/src/sub8_state_space.cc

```diff
@@ -0,0 +1,34 @@
+/**
+ * Author: Patrick Emami
+ * Date: 9/21/15
+ */
+
+#include "sub8_state_space.h"
+
+ompl::base::State* sub8::trajectory_generator::Sub8StateSpace::allocState() const {
+	StateType *state = new StateType();
```

Comment:
Why are you using time derivatives of Euler angles for angular velocity
instead of just the angular velocity vector? From your comments, it looks
like that's what you're actually doing already ("Get the angular velocity
about the y axis").

At least from what I've seen, all that does is introduce more complex
Jacobians everywhere and add singularities. It was only used in the old
trajectory generator because it worked directly on each Euler angle
separately, but AFAIK your approach won't.

On Wed, Sep 23, 2015 at 3:47 PM, Jake <notifications@github.com> wrote:

> In gnc/sub8_trajectory_generator/src/sub8_state_space.cc
> <https://github.com/uf-mil/Sub8/pull/20#discussion_r40248547>:
>
> > @@ -0,0 +1,34 @@
> > +/**
> > + * Author: Patrick Emami
> > + * Date: 9/21/15
> > + */
> > +
> > +#include "sub8_state_space.h"
> > +
> > +ompl::base::State* sub8::trajectory_generator::Sub8StateSpace::allocState() const {
> > +	StateType *state = new StateType();
>
> shared_ptr possible?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/uf-mil/Sub8/pull/20/files#r40248547>.
>